manager/integration/tests/test_stress.py

```diff
@@ -0,0 +1,264 @@
+        volume_name,
+        snapshot_name
+    )
+
+
+def delete_random_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+
+    snapshots_count = len(snapshots.data)
+
+    while True:
+        snapshot_id = randrange(0, snapshots_count)
+
+        if snapshots.data[snapshot_id].name == "volume-head":
+            continue
+        else:
+            break
+    snapshot_name = snapshots.data[snapshot_id].name
+
+    volume.snapshotDelete(name=snapshot_name)
+
+    trigger_purge = randrange(0, 2)
+
+    if trigger_purge == 1:
+        purge_random_snapshot(longhorn_api_client, volume_name, snapshot_name)
+
+
+@pytest.mark.stress
+def test_stress(generate_load):
+    pass
+
+
+@pytest.fixture(params=count)
+def generate_load(request):
+
+    index = request.param
+
+    longhorn_api_client = get_longhorn_api_client()
+    k8s_api_client = get_core_api_client()
+
+    volume_name = STRESS_VOLUME_NAME_PREFIX + index
+    pv_name = STRESS_PV_NAME_PREFIX + index
+    pvc_name = STRESS_PVC_NAME_PREFIX + index
+    pod_name = STRESS_POD_NAME_PREFIX + index
+
+    atexit.register(delete_and_wait_longhorn, longhorn_api_client, volume_name)
+    atexit.register(delete_and_wait_pv, k8s_api_client, pv_name)
+    atexit.register(delete_and_wait_pvc, k8s_api_client, pvc_name)
+    atexit.register(delete_and_wait_pod, k8s_api_client, pod_name)
+
+    longhorn_volume = create_and_check_volume(
+        longhorn_api_client,
+        volume_name,
+        size=VOLUME_SIZE
+    )
+
+    wait_for_volume_detached(longhorn_api_client, volume_name)
+
+    pod_manifest = generate_pod_with_pvc_manifest(pod_name, pvc_name)
+
+    create_pv_for_volume(longhorn_api_client,
+                         k8s_api_client,
+                         longhorn_volume,
+                         pv_name)
+
+    create_pvc_for_volume(longhorn_api_client,
+                          k8s_api_client,
+                          longhorn_volume,
+                          pvc_name)
+
+    create_and_wait_pod(k8s_api_client, pod_manifest)
+
+    write_data(k8s_api_client, pod_name)
+    create_snapshot(longhorn_api_client, volume_name)
+    write_data(k8s_api_client, pod_name)
+    create_snapshot(longhorn_api_client, volume_name)
+    delete_data(k8s_api_client, pod_name)
+    create_snapshot(longhorn_api_client, volume_name)
+
+    delete_random_snapshot(longhorn_api_client, volume_name)
+
+    # execute 3 more random actions
+    for round in range(3):
+        action = randrange(0, 4)
+
+        if action == 0:
+            write_data(k8s_api_client, pod_name)
+        elif action == 1:
+            delete_data(k8s_api_client, pod_name)
+        elif action == 2:
+            create_snapshot(longhorn_api_client, volume_name)
+        elif action == 3:
+            delete_random_snapshot(longhorn_api_client, volume_name)
+
+
+@pytest.mark.stress
+def test_reset_env():
```

Comment:
It's a bit unsafe to depends on the order of the test running. Do we have other ways to do it?

Reply:
I've kept `reset_env` test in case stress environment cleanup is needed.